core/src/main/scala/org/broadinstitute/dsde/rawls/bucketMigration/BucketMigrationService.scala

```diff
@@ -27,33 +30,173 @@ class BucketMigrationService(val dataSource: SlickDataSource, val samDAO: SamDAO
+
+      workspaces <- dataSource.inTransaction(_.workspaceQuery.listWithBillingProject(billingProject))
+      res <- op(workspaces)
+    } yield res
+
+  /**
+    * Entrypoints that make necessary authz checks then call relevant method to do the actual work
+    */
+  def adminGetBucketMigrationProgressForWorkspace(
+    workspaceName: WorkspaceName
+  ): Future[Option[MultiregionalBucketMigrationProgress]] =
+    asFCAdminWithWorkspace(workspaceName) {
+      getBucketMigrationProgressForWorkspaceInternal
+    }
+
+  def getBucketMigrationProgressForWorkspace(
+    workspaceName: WorkspaceName
+  ): Future[Option[MultiregionalBucketMigrationProgress]] =
+    asOwnerWithWorkspace(workspaceName) {
+      getBucketMigrationProgressForWorkspaceInternal
+    }
+
+  def adminGetBucketMigrationProgressForBillingProject(
+    billingProjectName: RawlsBillingProjectName
+  ): Future[Map[String, Option[MultiregionalBucketMigrationProgress]]] =
+    asFCAdminWithBillingProjectWorkspaces(billingProjectName) {
+      getBucketMigrationProgressForWorkspacesInternal
+    }
+
   def getBucketMigrationProgressForBillingProject(
     billingProjectName: RawlsBillingProjectName
   ): Future[Map[String, Option[MultiregionalBucketMigrationProgress]]] =
+    asOwnerWithBillingProjectWorkspaces(billingProjectName) {
+      getBucketMigrationProgressForWorkspacesInternal
+    }
+
+  def adminGetBucketMigrationAttemptsForWorkspace(
+    workspaceName: WorkspaceName
+  ): Future[List[MultiregionalBucketMigrationMetadata]] =
+    asFCAdminWithWorkspace(workspaceName) {
+      getBucketMigrationAttemptsForWorkspaceInternal
+    }
+
+  def getBucketMigrationAttemptsForWorkspace(
+    workspaceName: WorkspaceName
+  ): Future[List[MultiregionalBucketMigrationMetadata]] =
+    asOwnerWithWorkspace(workspaceName) {
+      getBucketMigrationAttemptsForWorkspaceInternal
+    }
+
+  def adminGetBucketMigrationAttemptsForBillingProject(
+    billingProjectName: RawlsBillingProjectName
+  ): Future[Map[String, List[MultiregionalBucketMigrationMetadata]]] =
+    asFCAdminWithBillingProjectWorkspaces(billingProjectName) {
+      getBucketMigrationAttemptsForWorkspacesInternal
+    }
+
+  def getBucketMigrationAttemptsForBillingProject(
+    billingProjectName: RawlsBillingProjectName
+  ): Future[Map[String, List[MultiregionalBucketMigrationMetadata]]] =
+    asOwnerWithBillingProjectWorkspaces(billingProjectName) {
+      getBucketMigrationAttemptsForWorkspacesInternal
+    }
+
+  def adminMigrateWorkspaceBucket(workspaceName: WorkspaceName): Future[MultiregionalBucketMigrationMetadata] =
+    asFCAdminWithWorkspace(workspaceName) {
+      migrateWorkspaceBucketInternal
+    }
+
+  def migrateWorkspaceBucket(workspaceName: WorkspaceName): Future[MultiregionalBucketMigrationMetadata] =
+    asOwnerWithWorkspace(workspaceName) {
+      migrateWorkspaceBucketInternal
+    }
+
+  def adminMigrateAllWorkspaceBuckets(
```

Comment:
I didn't bother abstracting out `asFCAdminWithWorkspaces` or `asOwnerWithWorkspaces` because these are currently the only methods that accept an arbitrary list of workspaces. If we end up going with this approach, I'll end up doing that when working on [WOR-1274](https://broadworkbench.atlassian.net/browse/WOR-1274)